### DISCUSSION TITLE
Umschaltung der Betriebsarten per mqtt geht nicht mehr

### DISCUSSION THREAD

**Discussion by Kenaschon** [dewenni/ESP_Buderus_KM271#48] [General]
Hallo,
wenn ich versuche, die Betriebsarten per mqtt umzuschalten, passiert nichts.

Per Web.Oberfläche geht es ohne Probleme.

Wo kann ich den Fehler wohl suchen?

Ich hatte den ESP neu geflasht.

Grüße
Manfred

**Comment by dewenni** [comment]
was genau sendest du per mqtt?

**Comment by Kenaschon** [comment]
mqtt.0.Heizung/setvalue/ww_betriebsart

mqtt.0.Heizung/setvalue/hk2_betriebsart

Es wird auch unter ioBroker angezeigt. Der Wert ändert sich entsprechend (0,1oder 2).

**Reply by dewenni** [reply]
als Payload schickst du auch 0, 1 oder 2?

**Comment by Kenaschon** [comment]
Je nachdem was ich benbötige. Beide bekommen "2" für Auomatik.

mqtt.0.Heizung/setvalue/ww_betriebsart eine 1 für Tagbetrieb

mqtt.0.Heizung.setvalue.hk2_betriebsart eine "0" für Nachtbetrieb.

Der Witz an der Sache ist. Es lief bis lang ohne Probleme. Dann bin ich von v3.2.3 auf v3.2.4 gegangen. Da lief es dann nicht mehr. Also alles wieder zurück auf v3.2.3.

Damit dort nichts verkehrtes über bleibt, habe ich den ESP32 vollständig gelöscht und die Version v3.2.3 neu aufgespielt. Ab da bekomme ich es nicht wieder zum laufen.

Per Web-Browser funktioniert alles zu 100%. Nur eben die Umschaltung per mqtt geht nicht mehr.

**Comment by Kenaschon** [comment]
Was kann ich noch probieren?

**Reply by dewenni** [reply]
Ich bin selbst noch nicht so richtig zum Testen gekommen.
Kann es sein, dass Mode 1 und 2 funktioniert und nir Mode 0 nicht?

**Reply by dewenni** [reply]
@Kenaschon bitte prüfe das Verhalten nochmal mit dem neuen Release 4.0.0

**Reply by Kenaschon** [reply]
Werde es versuchen.

Mit einem guten USB-Kabel klappt auch das flashen.

**Comment by dewenni** [comment]
Mi ist aber leider Gerde beim testen noch ein Fehler in der Anzeige aufgefallen.
Im Dashboard ist die Anzeige der Betriebsart vertauscht.
Korrigiere ich auch noch mit der nächsten Bugfix Version.

<img width="1399" alt="image" src="https://github.com/dewenni/ESP_Buderus_KM271/assets/46074831/a77998a1-6817-47fe-9e47-7fdd150b22a7">

**Reply by JieffeEl** [reply]
While correcting it, little English typo in language.h, line 25  {"Handbetrieb - Nacht", "Manual - Nacht"};

**Comment by Kenaschon** [comment]
Die neue Oberfläche sieht extrem gut aus. Vielen Dank dafür,

Was bei mir noch nicht funktioniert, ist das Importieren der json-Datei.

Die vertauschte Betriebsart betriff aber nur HK1, oder?

![400](https://github.com/dewenni/ESP_Buderus_KM271/assets/47242804/ad433f78-5f71-495c-a65f-be35c56fbd0d)

**Reply by dewenni** [reply]
Danke!
Ich muss mir das mit der vertauschten Betriebsart noch anschauen. Mache ich später.

Bezüglich config import. Wie äußert sich das Problem?
Ich habe es bei mir gerade nochmal getestet und es funktioniert.

**Reply by dewenni** [reply]
@Kenaschon kannst du mir noch Rückmeldung zu dem Thema mit dem Config Import geben?
Ansonsten würde ich noch ein Update fertig machen mit den anderen fixes.

**Comment by Kenaschon** [comment]
Kein Problem.

Wenn ich den ESP komplett neu flashe und dann versuche die Config zu laden, läd er sie zwar, aber er übernimmt die Einstellungen nicht. Ich kann das ruhig mehrmals machen. Ich muß trotzdem alle Einträge selber machen.

Ich hatte es jetzt auch noch nicht getestet ob er die Wlan Verbindung verliert wenn der MQTT-"Empfänger" nicht mehr da ist.


Das sind aber auch alles kleine unwichtige Problemchen. Bis jetzt läuft die Version 4.0.0. ohne Probleme.

**Comment by dewenni** [comment]
fixed with [release 4.0.1](https://github.com/dewenni/ESP_Buderus_KM271/releases/tag/v4.0.1)
